.github/workflows/docker.yaml

```diff
@@ -0,0 +1,56 @@
+name: Docker Build
+
+concurrency:
+  group: ${{github.repository}}-${{ github.ref_name }}-docker
+  cancel-in-progress: true
+
+on:
+  workflow_call:
+
+# Permissions needed
+# permissions:
+#   contents: read
+#   packages: write
+#   pull-requests: read
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          # list of Docker images to use as base name for tags
+          images: ghcr.io/${{github.repository}}
+          # generate Docker tags based on the following events/attributes
+          tags: |
+            type=edge,branch=main
+            type=ref,event=branch
+            type=ref,event=pr
+            type=semver,pattern={{version}}
+            type=semver,pattern={{major}}.{{minor}}
+            type=semver,pattern={{major}}
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Build and push
+        uses: docker/build-push-action@v5
+        with:
+          context: ./
+          file: .docker/Dockerfile
+          platforms: linux/amd64,linux/arm64
```

Comment:
let's expose context and file as input parameter (with default to current values), so that we can reuse this workflow also in repos where there are multiple images associated, e.g.: https://github.com/zaphiro-technologies/data-download-backend/blob/main/.github/workflows/docker.yaml